rpc.c

```diff
@@ -577,6 +578,7 @@ rpc_client_connect (rpc_instance rpc, const char *url)
         if (client->sock != NULL && !client->sock->dead)
         {
             /* This client will do */
+            client->timeout = timeout;
```

Comment:
This will persist until it is changed again, I don't think this is what you intended